otter/metrics.py

```diff
@@ -57,19 +60,35 @@ def get_tenant_metrics(tenant_id, scaling_groups, servers, _print=False):
     if _print:
         print('processing tenant {} with groups {} and servers {}'.format(
               tenant_id, len(scaling_groups), len(servers)))
-    metrics = []
-    for group in scaling_groups:
-        group_id = group['groupId']
-        create_metrics = partial(GroupMetrics, tenant_id,
-                                 group_id, group['desired'])
-        if group_id not in servers:
-            metrics.append(create_metrics(0, 0))
+
+    groups = {g['groupId']: g for g in scaling_groups}
+
+    def get_metric(args):
```

Comment:
this code got pretty complicated and I think it's because it's using merge_with when that's not really a good solution (I think it's only really useful when the data is relatively homogenous -- i.e. it doesn't matter which dict it comes from).

I think a simpler solution would be to iterate over `set(groups.keys() + servers.keys())`.